jwst/extract_1d/extract.py

```diff
@@ -2861,87 +2705,99 @@ def do_extract1d(input_model, ref_dict, smoothing_length=None,
             copy_keyword_info(slit, slit.name, spec)
             output_model.spec.append(spec)
     else:
-        # These default values for slitname are not really slit names, and
-        # slitname may be assigned a better value below, in the sections
-        # for input_model being an ImageModel or a SlitModel.
+        # These default values for slitname are not really slit names, and slitname may be assigned a better value
+        # below, in the sections for input_model being an ImageModel or a SlitModel.
         slitname = input_model.meta.exposure.type
+
         if slitname is None:
             slitname = ANY
+
         if slitname == 'NIS_SOSS':
             slitname = input_model.meta.subarray.name
 
         # Loop over these spectral order numbers.
         if input_model.meta.exposure.type == "NIS_SOSS":
-            # This list of spectral order numbers may need to be assigned
-            # differently for other exposure types.
+            # This list of spectral order numbers may need to be assigned differently for other exposure types.
             spectral_order_list = [1, 2, 3]
         else:
-            # For this case, we'll call get_spectral_order to get the order.
-            spectral_order_list = ["not set yet"]
+            spectral_order_list = ["not set yet"]  # For this case, we'll call get_spectral_order to get the order.
 
         if photom_has_been_run:
             pixel_solid_angle = input_model.meta.photometry.pixelarea_steradians
+
             if pixel_solid_angle is None:
-                log.warning("Pixel area (solid angle) is not populated; "
-                            "the flux will not be correct.")
+                log.warning("Pixel area (solid angle) is not populated; the flux will not be correct.")
                 pixel_solid_angle = 1.
         else:
-            pixel_solid_angle = 1.                      # not needed
+            pixel_solid_angle = 1.  # not needed
 
         source_type = input_model.meta.target.source_type
-        input_units_are_megajanskys = (photom_has_been_run and
-                                       source_type == 'POINT' and
-                                       (instrument == 'NIRSPEC' or
-                                        exp_type == 'NIS_SOSS'))
+
+        input_units_are_megajanskys = (
```

Comment:
This logic seems dangerous. If at some point it's decided to change the calibrated units for one of these modes, someone has to make sure to update this code. Wouldn't it be better to just rely on the BUNIT keyword value? Of course if we're only making style updates here, I guess that's not an appropriate change.

Reply:
I agree, there's quite a lot of refactoring that needs to be done, this included. But, yes, I think that that should go in a separate "refactor extract 1d" PR